papas/mergedclusterbuilder.py

```diff
@@ -61,28 +61,18 @@ def __init__(self, clusters, ruler, history_nodes = None):
         
         #make sure we use the original history and update it as needed
         self.history_nodes = history_nodes
+        self._make_and_store_merged_clusters()
 
-        self._make_merged_clusters()
-
-    def _make_merged_clusters(self):
+    def _make_and_store_merged_clusters(self):
```

Comment:
In cpp I originally separated making objects (such as a merged cluster) and storing them. This was not v convenient when I changed the unique id to be non static because one needs to know about the collection into which the new object is to be added (so as to make the unique id). So I changed the way it worked and the function names in C++ to reflect this and also here in python.